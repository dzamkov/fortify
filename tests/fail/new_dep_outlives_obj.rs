use fortify::*;

fn main() {
    let x = Box::new(123);
    let obj = &*x;
    let fortified = Fortify::new_dep(obj, |val| Lowered::new(*val));
    drop(x);
    assert_eq!(**fortified.borrow(), 123);
}