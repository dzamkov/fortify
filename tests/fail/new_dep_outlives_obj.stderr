error[E0505]: cannot move out of `x` because it is borrowed
 --> tests/fail/new_dep_outlives_obj.rs:7:10
  |
5 |     let obj = &*x;
  |               --- borrow of `*x` occurs here
6 |     let fortified = Fortify::new_dep(obj, |val| Lowered::new(*val));
7 |     drop(x);
  |          ^ move out of `x` occurs here
8 |     assert_eq!(**fortified.borrow(), 123);
  |                  ------------------ borrow later used here
